sdks/keyvault/src/client.rs

```diff
@@ -15,109 +12,43 @@ pub(crate) const API_VERSION: &str = "7.0";
 ///
 /// ```no_run
 /// use azure_keyvault::KeyVaultClient;
-/// let client = KeyVaultClient::new(&"{client_id}", &"{client_secret}", &"{tenant_id}", &"test-keyvault");
+/// use azure_auth_aad::DefaultCredential;
+/// let creds = DefaultCredential::default();
+/// let client = KeyVaultClient::new(&creds, &"test-keyvault");
 /// ```
 #[derive(Debug)]
-pub struct KeyVaultClient<'a> {
-    pub(crate) aad_client_id: &'a str,
-    pub(crate) aad_client_secret: &'a str,
-    pub(crate) aad_tenant_id: &'a str,
+pub struct KeyVaultClient<'a, T> {
+    pub(crate) token_credential: &'a T,
     pub(crate) keyvault_name: &'a str,
     pub(crate) endpoint_suffix: String,
     pub(crate) keyvault_endpoint: String,
-    pub(crate) token: Option<AccessToken>,
-    pub(crate) token_expiration: Option<DateTime<Utc>>,
+    pub(crate) token: Option<TokenResponse>,
 }
 
-impl<'a> KeyVaultClient<'a> {
+impl<'a, T: TokenCredential> KeyVaultClient<'a, T> {
     /// Creates a new `KeyVaultClient` with an endpoint suffix. Useful for non-public Azure clouds.
     /// For the default public environment, use `KeyVaultClient::new`.
     ///
     /// # Example
     ///
     /// ```no_run
     /// use azure_keyvault::KeyVaultClient;
-    /// let client = KeyVaultClient::with_endpoint_suffix(&"c1a6d79b-082b-4798-b362-a77e96de50db", &"SUPER_SECRET_KEY", &"bc598e67-03d8-44d5-aa46-8289b9a39a14", &"test-keyvault", "vault.azure.net".to_owned());
+    /// use azure_auth_aad::DefaultCredential;
+    /// let creds = DefaultCredential::default();
+    /// let client = KeyVaultClient::with_endpoint_suffix(&creds, &"test-keyvault", "vault.azure.net".to_owned());
     /// ```
     pub fn with_endpoint_suffix(
-        aad_client_id: &'a str,
-        aad_client_secret: &'a str,
-        aad_tenant_id: &'a str,
+        token_credential: &'a T,
         keyvault_name: &'a str,
         endpoint_suffix: String,
```

Comment:
I know this was like this before, but since we're feeding this into a format! macro, it seems more wise to take `endpoint_suffix` as a `&str`.

Reply:
Same as above, original code. I don't know about original intents.

Reply:
I agree, an oversight on my end.